### PR TITLE
Implement controller logic for VerticaRestorePointsQuery Part 2

### PR DESCRIPTION
**Pull request by chinhtranvan**
This PR represents the second piece of implementing controller logic for VerticaRestorePointsQuery. It involves copying the result of the query into the status fields of VerticaRestorePointsQuery CR

### CODE REVIEW

File: pkg/vrpqstatus/status.go
Comment (spilchen) [inline pkg/vrpqstatus/status.go:90]: Do we have any use cases of callers external to this package calling Update? I wonder if UpdateStatusForRestorePointsQuery should be renamed to Update, and Update renamed to updateImpl.

File: pkg/vrpqstatus/status.go
Comment (spilchen) [inline pkg/vrpqstatus/status.go:82]: If we can use the actual RestorePoints struct from vclusterops here, then we don't need this loop. It should be a straight copy.

File: pkg/vadmin/restore_points_query_at.go
Comment (spilchen) [inline pkg/vadmin/restore_points_query_at.go:27]: You can return `nil` for the first parameter.

File: pkg/vadmin/restore_points_query_vc_test.go
Comment (spilchen) [inline pkg/vadmin/restore_points_query_vc_test.go:75]: Could we update the unit test so that we do return something that we can check?

File: tests/e2e-leg-6/verify-vrpq-conditions/10-assert.yaml
Comment (spilchen) [inline tests/e2e-leg-6/verify-vrpq-conditions/10-assert.yaml:45]: You can combine this into the one `VerticaRestorePointsQuery`.

File: tests/e2e-leg-6/verify-vrpq-conditions/04-create-archive-and-restorepoints.yaml
Comment (spilchen) [inline tests/e2e-leg-6/verify-vrpq-conditions/04-create-archive-and-restorepoints.yaml:17]: These would be better suited if everything went into a configMap/pod and run there. See tests/e2e-leg-3/client-access/35-verify-memory-limit.yaml.

File: api/v1beta1/verticarestorepointsquery_types.go
Comment (chinhtranvan) [reply api/v1beta1/verticarestorepointsquery_types.go:64]: There is one thing that prevent that, temporarily, RestorePoint defined in the vclusterOps package missing Json tag
e.g,
 ```
 Archive string `json:"archive"`
 ID string `json:"id"`
```
We need to open a PR for vcluster side in the server side to update json, ortherwise, Go complains "/home/chtran/go/pkg/mod/github.com/vertica/vcluster@v1.1.1-0.20240130155647-29203c00b7b5/vclusterops/nma_show_restore_points_op.go:109:2: encountered struct field "Archive" without JSON tag in type "RestorePoint"
/home/chtran/go/pkg/mod/github.com/vertica/vcluster@v1.1.1-0.20240130155647-29203c00b7b5/vclusterops/nma_show_restore_points_op.go:111:2: encountered struct field "ID" without JSON tag in type "RestorePoint"
/home/chtran/go/pkg/mod/github.com/vertica/vcluster@v1.1.1-0.20240130155647-29203c00b7b5/vclusterops/nma_show_restore_points_op.go:114:2: encountered struct field "Index" without JSON tag in type "RestorePoint"
/home/chtran/go/pkg/mod/github.com/vertica/vcluster@v1.1.1-0.20240130155647-29203c00b7b5/vclusterops/nma_show_restore_points_op.go:116:2: encountered struct field "Timestamp" without JSON tag in type "RestorePoint""

Comment (spilchen) [reply api/v1beta1/verticarestorepointsquery_types.go:64]: I'm okay with adding the JSON tags.

Comment (chinhtranvan) [reply api/v1beta1/verticarestorepointsquery_types.go:64]: This is the PR from sever side
https://git.verticacorp.com/projects/VER/repos/server/pull-requests/20194/overview
I already address all your comments, just wait until this PR is merged, then I can push the changes

**Review by spilchen — APPROVED** [review]

**Pull request merged**
